### PR TITLE
Documentation infrastructure.

### PR DESCRIPTION
**Pull request by jezdez**
This does a few things:

- [x] Adds some placeholders following the Diátaxis framework (https://diataxis.fr)
- [x] Sets up Sphinx with MyST parser for Markdown in addition to rST.
- [x] Uses the well-known [PyData Sphinx theme](https://github.com/pydata/pydata-sphinx-theme).
- [x] Moves some already existing Markdown files into the docs directory.
- [x] Sets up the initial doc review GitHub action to auto-build
- [x] @ximena9201 Autodeploys to S3 for long-term deployment
- [ ] @jezdez Set up Markdown linting using pre-commit (on hold because of https://github.com/executablebooks/mdformat-myst/pull/9)
- [x] @jezdez Replace sphinx-panels with sphinx-design
- [x] @jezdez Add sphinx-autobuild for easier local editing
- [x] add redirect from https://docs.pyscript.net/ to https://docs.pyscript.net/latest/
- [ ] Add link to HTML Editor
- [ ] Copy/paste/run of example code in HTML editor
- [ ] Add overview of PyScript tags for Getting Started page
- [ ] Add a funny error.html file to the S3 bucket so 404 errors are handled appropriately
- [ ] Version selection between tags
- [x] Keep governance docs in the root of repo

### Deployment (like ReadTheDocs)

Eventually we need three different types of deployments:

- [x] review (pushed to `docs.pyscript.net/review/<pr-number>/`)
- [x] tag (pushed to `docs.pyscript.net/<tag>/`)
- [x] latest (pushed to `docs.pyscript.net/latest/`)

Refs #167 and #168.

### Known issues

- the "Edit this page" link is not working on https://docs.pyscript.net/review/262/ since that points to the main branch which doesn't have the files yet, because this PR adds them

### CODE REVIEW

File: README.md
Comment (kathatherine) [inline README.md:8]: These introductory phrases: "To get started" and "For examples" need commas after them.

File: docs/concepts/governance/policy.md
Comment (kathatherine) [inline docs/concepts/governance/policy.md:3]: "Project policies"? Typo there.

File: MAINTAINERS.md
Comment (philippjfr) [inline MAINTAINERS.md:1]: Was this removal intentional?

Comment (philippjfr) [reply MAINTAINERS.md:1]: Guess it's duplicated in the governance repo.

File: README.md
Comment (idanenglander) [inline README.md:6]: Add comma after "JSFiddle"
Remove comma after "hackable"

File: docs/README.md
Comment (idanenglander) [inline docs/README.md:4]: Either delete "is" or try this rewrite:

Welcome to the PyScript documentation directory, where you can find and contribute to discussions around PyScript and related topics.

File: docs/README.md
Comment (idanenglander) [inline docs/README.md:16]: Minor adjustments. The phrase "such as templates and themes" could alternatively be between commas or parentheses:
The `docs` directory in the pyscript repository contains the
[Sphinx](https://www.sphinx-doc.org/) documentation project, a system for publishing static files—such as templates and themes—and plaintext files written in markdown.

File: docs/concepts/governance/maintainers.md
Comment (idanenglander) [inline docs/concepts/governance/maintainers.md:3]: "represent" isn't really working grammatically here, though I understand the goal of it... perhaps:

"... you authorize the binding of that organization to these policies."

File: docs/tutorials/index.md
Comment (idanenglander) [inline docs/tutorials/index.md:3]: typo in tutorials

File: docs/README.md
Comment (kathatherine) [reply docs/README.md:16]: Ooh, yes, the em dashes read better.

File: docs/tutorials/index.md
Comment (cclauss) [reply docs/tutorials/index.md:3]: #263 should automatically flag these for us.

File: docs/README.md
Comment (kathatherine) [reply docs/README.md:16]: Did a slightly different rewrite. Let me know what you think once I push my changes.

File: docs/concepts/governance/maintainers.md
Comment (kathatherine) [reply docs/concepts/governance/maintainers.md:3]: Since @kevingoldsmith put all these governance docs together, I'd like his input on that change before we make it.

**Comment by kathatherine** [comment]
@fpliger @jezdez I also did some general edits and replaced the Lorem Ipsum with some "Coming Soon" text anywhere it was needed. Idan, Albert, and I added some content to the Concepts folder to attempt to give a big picture explanation of PyScript. It still needs work, but is at least mergeable, if y'all are okay with the language.

**Comment by ximena9201** [comment]
@jezdez I've added some changes in this [GHA](https://github.com/pyscript/pyscript/pull/262/files#diff-d9f3bcc802c05529274279e546ef4729bb14ef96d075896caf526fff5240175c) that puts the docs/ ** content to a S3 public bucket.  I guess we still need to automate the releases, tags, or probably make some slightly changes.

**Comment by cclauss** [comment]
LGTM...  Nice work.  Merge conflict to look into.  Py-script is delivering Python 3.10 but several of these GHAs use Py3.9.  There might be plenty of good reasons for this dichotomy but it might lead to issues.

There is probably some the thought to be given to "_There is no setup_" vs. docs/tutorials/setup.md

**Review by cclauss — APPROVED** [review]

**Comment by jezdez** [comment]
> @jezdez I've added some changes in this [GHA](https://github.com/pyscript/pyscript/pull/262/files#diff-d9f3bcc802c05529274279e546ef4729bb14ef96d075896caf526fff5240175c) that puts the docs/ ** content to a S3 public bucket. I guess we still need to automate the releases, tags, or probably make some slightly changes.

I'll working on that tomorrow if that's okay, thank you for adding those changes!

**Comment by jezdez** [comment]
> LGTM... Nice work. Merge conflict to look into. Py-script is delivering Python 3.10 but several of these GHAs use Py3.9. There might be plenty of good reasons for this dichotomy but it might lead to issues.

Good callout, let's fix the Python version difference, no good reason for it IIRC. 

> There is probably some the thought to be given to "_There is no setup_" vs. docs/tutorials/setup.md

Huh, fascinating, while I like the messaging of "there is no setup", a tutorial would probably require a user to setup a development environment, to make sure they can actually edit a HTML file and use PyScript. That's worth doing to engage with beginners that will look for ways to start using PyScript. I wonder if we could look at "web development" tutorials out there (MDN?) and get inspiration.

**Comment by jezdez** [comment]
@fpliger So generally speaking, it would be interesting to understand what would be a good level at which we should do a review and merge of this PR? This is still just a skeleton in terms of content and design, but it might be worth merging to allow smaller content/design PRs to happen.

https://docs.pyscript.net/review/262/ is the current build of this PR (as will all PRs automatically trigger such a review build).

**Comment by fpliger** [comment]
@jezdez totally agree. This became too big of a catch all PR. If you are 👍  I'd say, I can review and merge this and the allow for other smaller (and more digestible) PR to follow. Sounds good to you?

**Comment by jezdez** [comment]
> @jezdez totally agree. This became too big of a catch all PR. If you are 👍 I'd say, I can review and merge this and the allow for other smaller (and more digestible) PR to follow. Sounds good to you?

Sounds good to me! Thank you <3

File: docs/concepts/governance/maintainers.md
Comment (kevingoldsmith) [reply docs/concepts/governance/maintainers.md]: represent comes from the original github document. It is meant to be legalese. I wouldn't change it if it is a bit janky but the meaning is clear.

File: MAINTAINERS.md
Comment (kevingoldsmith) [inline MAINTAINERS.md:1]: @jezdez is this correct to move the governance stuff from the top of the repo into a subdirectory of the docs directory?

File: docs/concepts/governance/maintainers.md
Comment (kevingoldsmith) [inline docs/concepts/governance/maintainers.md:1]: I'm not sure we should be burying these files.

**Review by kevingoldsmith — CHANGES_REQUESTED** [review]
generally looks ok, but I'm concerned about making some of the governance documents undiscoverable. I think they make more sense at the project level than buried in the docs. If we want them in the docs we can either reference them at the room, link to them or rephrase them.

File: docs/concepts/governance/maintainers.md
Comment (fpliger) [reply docs/concepts/governance/maintainers.md:1]: I'm +1 on bumping it for governance. In my mind, it's a "special" concept that we want to give big priority.

On a different note, it'd be great if we could link/pull these directly from the governance repo.

**Review by fpliger — APPROVED** [review]
Wow.. this is an AMAZING starting point!! It'd be +100 on merging this right now. 

On a quick chat with @jezdez  we agreed that it makes sense to add something like "THIS IS A PLACEHOLDER FOR THE DOCUMENTATION, DO NOT USE IT AS A REFERENCE YET" to the front page until we add meaningful content. 

Will merge as soon as it lands :)

File: docs/concepts/governance/maintainers.md
Comment (jezdez) [reply docs/concepts/governance/maintainers.md:1]: Oh, good point @kevingoldsmith, I'll move them back and use a symlink to have the real files in the root of the repo, but also render them on the docs site (we can make a editorial decision if we want to keep them separately).

@fpliger FWIW this structure was recommended by the [MVG](https://github.com/github/MVG), to have some files in the project repo, and some on the org level. So if we were to add additional projects to the pyscript organization, they would require also at least a `MAINTAINERS.md` and `GOVERNANCE.md`.

File: docs/conf.py
Comment (jezdez) [inline docs/conf.py:22]: @kevingoldsmith Is that copyright line correct?

File: MAINTAINERS.md
Comment (jezdez) [reply MAINTAINERS.md:1]: No, you're right, reverting.

File: docs/concepts/governance/maintainers.md
Comment (jezdez) [inline docs/concepts/governance/maintainers.md:1]: @kevingoldsmith Reverted to just symlink to the root of the repo, while still [rendering them in the docs](https://docs.pyscript.net/review/262/concepts/governance/maintainers.html).

**Comment by jezdez** [comment]
@fpliger I've addressed @kevingoldsmith's comments and add the note, does that work?

CleanShot 2022-05-16 at 23 43 29@2x

**Comment by jezdez** [comment]
When the PR is merged, I'll open new issues for the TODO items above left undone. I think the are fine to work on after the fact.

**Comment by fpliger** [comment]
Imo, that works great! I'm +1 on merging this as is and opening issues/PRs for the remaining items /comments (there's a lot of value here that others can build on top of). @kevingoldsmith does that work for you or do you have any reservations?

**Comment by fpliger** [comment]
Thank you for the great work on this @jezdez @ximena9201 @kathatherine @idanenglander and everyone involved!

**Pull request merged**
